File: java/gigasecond/src/main/java/Gigasecond.java
import java.time.LocalDate;
import java.time.LocalDateTime;

class Gigasecond {

    private LocalDateTime moment;
    private static long GIGASECOND = 1_000_000_000;

    Gigasecond(LocalDate moment) {
        this.moment = moment.atStartOfDay();
    }

    Gigasecond(LocalDateTime moment) {
        this.moment = moment;
    }

    LocalDateTime getDateTime() {
        return moment.plusSeconds(GIGASECOND);
    }

}

File: kotlin/gigasecond/src/main/kotlin/Gigasecond.kt
import java.time.LocalDate
import java.time.LocalDateTime
import java.time.LocalTime

class Gigasecond(private val baseDateTime: LocalDateTime) {

    constructor(baseDate: LocalDate) : this(LocalDateTime.of(baseDate, LocalTime.MIN))

    val date: LocalDateTime = baseDateTime.plusSeconds(1_000_000_000)
}
